taurus/entity/dict_serializable.py

```diff
@@ -82,7 +83,21 @@ def build(d):
         return loads(dumps(d))
 
     def __repr__(self):
-        return str(self.as_dict())
+        from taurus.util.impl import substitute_links, set_uuids
+        object_dict = self.as_dict()
+        # as_dict skips over keys in `skip`, but they should be in the representation.
+        skipped_keys = {x.lstrip('_') for x in vars(self) if x in self.skip}
+        for key in skipped_keys:
+            skipped_attribute = deepcopy(self.__getattribute__(key))
+            # Replace links in skipped keys with LinkByUID to prevent infinite recursion loop.
+            set_uuids(skipped_attribute)
```

Comment:
I'm a little worried about a mutating an object in a display function like this, even if it's on a deep copy of an object. Since this is a "soft link" anyway, can you just make the skipped items show up as links instead of replacing their children with links? Open to other suggestions, too.

Reply:
I guess there's no guarantee that the children will have an id yet. I will think about this some more.

Reply:
The impetus here is that when a user displays an object in a Jupyter notebook they see `__repr__()`. And the idea of a "soft link" is not relevant to the user. From their point of view the material run has measurements. So if they see that the `measurements` field in the material run is missing (what happens now), or they just see a list of LinkByUID, it gives them the sense that the measurements are not there/have been turned into links. That's why I wanted to make the actual object appear in `__repr__`.

Reply:
> I guess there's no guarantee that the children will have an id yet. I will think about this some more.
In Citrine-Python we automatically generate IDs as soon as every object is created, so this will never be invoked. But in Taurus there's no guarantee that the object has uids.

Reply:
Well the original context was just that people thought they were literally just missing. I think the user will realize they can just print out `material.measurements` as long as `measurements` should up at all in the output. We don't need to recreate the whole tree here, they can dump JSON if they need that.

Reply:
I think `measurements: <List[MeasurementRun]>` would probably be sufficient to convince the user that their data is there.

Reply:
That's reasonable, but I'd like to run it by strike-team-data before committing to anything.